### ISSUE TITLE
Flair + Huggingface: finetuning classifier or whole model?

### ISSUE THREAD

**Issue by Flowhill** [flairNLP/flair#2934]
I am unsure as to what weights are updated when training the following pipeline:
As a TransformerDocumentEmbeddings I use a huggingface model such as 'xlm-roberta-base'.
A CSVClassificationCorpus is used with local files as a dataset and a label dictionary is created from it.
A TextClassifier is added with the 'xlm-roberta-base' "embeddings model".
Finally, a ModelTrainer is given the corpus and the classifier after which finetuning starts.

During finetuning are the weights in the roberta model frozen and are we only adjusting the classifier weights or are all weights (model + classifier) trained?

Do we have any insight into what kind of layers the classifier network consists of?

**Comment by Flowhill** [comment]
Update:
When finetuning the code outputs the whole model structure e.g.
```
2022-09-09 16:58:52,068 Model: "TextClassifier(
  (decoder): Linear(in_features=768, out_features=18, bias=True)
  (dropout): Dropout(p=0.0, inplace=False)
  (locked_dropout): LockedDropout(p=0.0)
  (word_dropout): WordDropout(p=0.0)
  (loss_function): BCEWithLogitsLoss()
  (document_embeddings): TransformerDocumentEmbeddings(
    (model): XLMRobertaModel(
```

It leaves the question whether the weights of the model (XLMRoberta) are automatically frozen and only the Linear decoder is trained.

**Comment by lukasgarbas** [comment]
You can control this by setting fine_tune parameter to true/false in ```TransformerDocumentEmbeddings(fine_tune=True)```. By default, fine-tuning is set to true, which means that the transformer weights and the text classifier (a single linear layer as the decoder) will be adjusted. You can also train just the classifier by setting fine-tuning to false. However, this might need a different learning rate as compared to fine-tuning. Hope this helps!
